src/gc.c

```diff
@@ -296,6 +296,7 @@ static void gc_allocation_map_remove(AllocationMap* am,
     Allocation* cur = am->allocs[index];
     Allocation* prev = NULL;
     while(cur != NULL) {
+        Allocation* cur_next = cur->next;
```

Comment:
🤦🏼‍♂️ ... nice catch, though! 
Can we (a) rename `cur_next` to `next` and (b) move the declaration above the loop for consistency?

Reply:
done